### PR TITLE
Add API function for getting last deployed sysroot

### PR DESCRIPTION
**Pull request by OYTIS**
While not used by OSTree, this function can be useful to other users of libostree (like https://github.com/advancedtelematic/aktualizr) as a good approximation of what deployment we're going to boot next after the reboot.

### CODE REVIEW

File: src/libostree/ostree-sysroot.c [hunk 1]
```diff
@@ -1132,23 +1164,9 @@ ostree_sysroot_get_merge_deployment (OstreeSysroot     *self,
...
-          OstreeDeployment *deployment = self->deployments->pdata[i];
-
-          if (strcmp (ostree_deployment_get_osname (deployment), osname) != 0)
-            continue;
-
-          return g_object_ref (deployment);
-        }
-    }
-  return NULL;
+      return ostree_sysroot_get_last_deployed(self, osname);
```
Comment (jlebon) [inline src/libostree/ostree-sysroot.c]: Minor nit: spacing between `(`.

File: src/libostree/ostree-sysroot.c [hunk 2]
```diff
@@ -1108,6 +1108,38 @@ find_booted_deployment (OstreeSysroot       *self,
...
+ * ostree_sysroot_get_last_deployed:
+ * @self: Sysroot
+ * @osname: (allow-none): Operating system group
+ *
+ * Find the last deployed sysroot.
+ *
+ * Returns: (transfer full): The last deployment in OS or NULL if none
+ */
+OstreeDeployment *
+ostree_sysroot_get_last_deployed (OstreeSysroot     *self,
```
Comment (jlebon) [inline src/libostree/ostree-sysroot.c]: I'd call this something like `ostree_sysroot_get_default_deployment`?

Actually, we have a similar version of this in rpm-ostree that returns not just the pending deployment, but also the rollback:

https://github.com/projectatomic/rpm-ostree/blob/95e9aa4/src/daemon/rpmostree-sysroot-core.c#L399

Maybe we should just upstream that to libostree.

Comment (cgwalters) [inline src/libostree/ostree-sysroot.c]: Makes sense to me, can you try that @OYTIS  ?  `ostree_sysroot_query_deployments` ?

**Comment by cgwalters** [comment]
Pushed a rewritten version for this :arrow_up:

**Comment by cgwalters** [comment]
@rh-atomic-bot delegate=OYTIS

**Comment by OYTIS** [comment]
Thanks, looks great.

**Comment by OYTIS** [comment]
@rh-atomic-bot r+ b082119

**Pull request closed without merging**